packages/contracts-bedrock/src/universal/OptimismMintableERC20FactoryInterop.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { OptimismMintableERC20Factory, OptimismMintableERC20 } from "src/universal/OptimismMintableERC20Factory.sol";
+import { IOptimismERC20Factory } from "src/L2/IOptimismERC20Factory.sol";
+import { CREATE3 } from "@rari-capital/solmate/src/utils/CREATE3.sol";
+
+/// @custom:proxied
+/// @custom:predeployed 0x4200000000000000000000000000000000000012
+/// @title OptimismMintableERC20FactoryInterop
+/// @notice OptimismMintableERC20FactoryInterop is a factory contract that generates OptimismMintableERC20
+///         contracts on the network it's deployed to, using CREATE3. Simplifies the deployment process for users
+///         who may be less familiar with deploying smart contracts. Designed to be backwards
+///         compatible with the older OptimismMintableERC20Factory contract.
+contract OptimismMintableERC20FactoryInterop is OptimismMintableERC20Factory, IOptimismERC20Factory {
+    /// @notice Storage slot that the OptimismMintableERC20FactoryInteropStorage struct is stored at.
+    /// keccak256(abi.encode(uint256(keccak256("optimismMintableERC20FactoryInterop.storage")) - 1)) &
+    /// ~bytes32(uint256(0xff));
+    bytes32 internal constant OPTIMISM_MINTABLE_ERC20_FACTORY_INTEROP_SLOT =
+        0xb6fdf24dfda35722597f70f86628ef4ce6db853b20879bdd2d61f0f5d169b100;
+
+    /// @notice Storage struct for the OptimismMintableERC20FactoryInterop storage.
+    /// @custom:storage-location erc7201:optimismMintableERC20FactoryInterop.storage
+    struct OptimismMintableERC20FactoryInteropStorage {
+        /// @notice Mapping of local token address to remote token address.
+        mapping(address => address) deployments;
```

Comment:
I've seen that you implemented it only here, but do we also need it on the `OptimismMintableERC20Factory` file?

Reply:
As per this spec, yes, but not completely sure if something has changed 
https://github.com/ethereum-optimism/specs/pull/294/files#diff-93042f57f61e17a3b53f6177cce570383fd0e91501a9b0665608dec89c962519R148

Reply:
The thing is that this `OptimismMintableERC20FactoryInterop` is the new implementation for the factory predeploy so it will have it anyway. The proxy will point to this new contract that inherits the old one.

Reply:
ahh perfect then!